packages/eslint-plugin-next/lib/rules/next-script-for-ga.js

```diff
@@ -0,0 +1,61 @@
+const NodeAttributes = require('../utils/node-attributes.js')
+
+const GA_URL = 'www.google-analytics.com/analytics.js'
+const ERROR_MSG =
+  'Use the Script component for loading third party scripts. See: https://nextjs.org/docs/messages/next-script-for-ga.'
+
+module.exports = {
+  meta: {
+    docs: {
+      description:
+        'Prefer next script component when using the inline script for Google Analytics',
+      recommended: true,
+    },
+  },
+  create: function (context) {
+    return {
+      JSXOpeningElement(node) {
+        if (node.name.name !== 'script') {
+          return
+        }
+        if (node.attributes.length === 0) {
+          return
+        }
+        const attributes = new NodeAttributes(node)
+
+        // Check if the Alternative async tag is being used to add GA. https://developers.google.com/analytics/devguides/collection/analyticsjs#alternative_async_tag
+        if (
+          typeof attributes.value('src') === 'string' &&
+          attributes.value('src').includes(GA_URL)
+        ) {
+          return context.report({
+            node,
+            message: ERROR_MSG,
+          })
+        }
+
+        // Check if inline script is being used to add GA. https://developers.google.com/analytics/devguides/collection/analyticsjs#the_google_analytics_tag
+        if (
+          attributes.has('dangerouslySetInnerHTML') &&
+          attributes.value('dangerouslySetInnerHTML')[0]
+        ) {
+          const htmlContent =
+            attributes.value('dangerouslySetInnerHTML')[0].value.quasis &&
+            attributes.value('dangerouslySetInnerHTML')[0].value.quasis[0].value
+              .raw
+          if (
+            htmlContent &&
+            htmlContent.includes('www.google-analytics.com/analytics.js')
```

Comment:
```suggestion
            htmlContent.includes(GA_URL)
```
Should we test more URLs related to Google analytics? It seems our examples currently use `googletagmanager` [here](https://github.com/vercel/next.js/blob/d2551bbbc7588d392fd47420c4eea803794f967f/examples/with-google-analytics/pages/_document.js#L13) and [here](https://github.com/vercel/next.js/blob/d2551bbbc7588d392fd47420c4eea803794f967f/examples/with-google-tag-manager/pages/_document.js#L25)